src/db.rs

```diff
@@ -85,6 +141,83 @@ pub fn insert_connection(connection: Connection) -> Result<Uuid> {
     Ok(nwuuid)
 }
 
+pub fn insert_policy(policy: &policy::Policy) -> Result<i64> {
+    let mut dbconn = get_dbconn().unwrap();
+
+    let mysqlstr = "INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, 
+                    allowed_build_ids, create_date, valid)
+                    VALUES(?, ?, ?, ?, ?, NOW(), 1)".to_string();
+
+    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    /* create JSON for vector struct member variables */
+    let adj = serde_json::to_string(&policy.allowed_digests)?;
+    let apj = serde_json::to_string(&policy.allowed_policies)?;
+    let abidj = serde_json::to_string(&policy.allowed_build_ids)?;
+
+    trnsx.exec_drop(
+        mysqlstr,
+        (
+            &adj,
+            &apj,
+            policy.min_fw_api_major,
+            policy.min_fw_api_minor,
+            &abidj,
+        ),
+    )?;
+
+    let lid = trnsx.query_first("SELECT LAST_INSERT_ID()")?.unwrap();
+    trnsx.commit()?;
+
+    Ok(lid)
+}
+
+pub fn get_policy(pid: i64) -> Result<policy::Policy, anyhow::Error> {
+    let mut dbconn = get_dbconn().unwrap();
+
+    let pidstr = pid.to_string();
+    let mqstr = "SELECT allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids, create_date, delete_date, valid  FROM policy WHERE id = ?";
+
+    let qrow: mysql::Row = dbconn.exec_first(mqstr, (pidstr,))?.unwrap();
+
+    let adval = mysql::from_value_opt::<String>(qrow["allowed_digests"].clone())?;
+
+    let cmajval = mysql::from_value_opt::<i64>(qrow["min_fw_api_major"].clone())?;
+
+    let cminval = mysql::from_value_opt::<i64>(qrow["min_fw_api_minor"].clone())?;
+
+    let acapval = mysql::from_value_opt::<String>(qrow["allowed_policies"].clone())?;
+
+    let abival = mysql::from_value_opt::<String>(qrow["allowed_build_ids"].clone())?;
+
+    let cvalid = mysql::from_value_opt::<i64>(qrow["valid"].clone())?;
```

Comment:
Maybe you could add a `WHERE` clause to the query so that you only select valid policies. Then you wouldn't have to worry about handling validity here.

Reply:
I updated get policy to include an AND on valid to only select valid policies and simplified the return.